analyses/infercnv-consensus-cell-type/scripts/01_run-infercnv.R

```diff
@@ -0,0 +1,211 @@
+  "reference_file does not exist" = file.exists(opts$reference_file),
+  "gene_order_file does not exist" = file.exists(opts$gene_order_file),
+  "annotation_file was not specified" = !is.null(opts$annotation_file),
+  "output_dir was not specified" = !is.null(opts$output_dir),
+  "scratch_dir was not specified" = !is.null(opts$scratch_dir),
+  "hmm_model not properly specified" = opts$hmm_model %in% c("i3", "i6")
+)
+
+# Setup files and paths -----------------
+set.seed(opts$seed)
+
+# read SCE file so we can set up paths with its library id
+sce <- readRDS(opts$sce_file)
+
+# define reference name to set up output files and paths
+ref_name <- stringr::str_remove(
+  basename(opts$reference_file),
+  "^ref-"
+) |>
+  stringr::str_remove("\\.rds$")
+
+# get library id and construct scratch directory
+library_id <- metadata(sce)$library_id
+
+# create output and scratch directory if not already present for library id
+fs::dir_create(opts$output_dir)
+
+# organize scratch by library id
+scratch_dir <- file.path(opts$scratch_dir, library_id)
+fs::dir_create(scratch_dir)
+
+# define output metadata file
+cnv_metadata_file <- file.path(opts$output_dir, glue::glue("{library_id}_reference-{ref_name}_cnv-metadata.tsv"))
+# define path to output cnv metadata file saved in scratch dir
+scratch_metadata_file <- file.path(scratch_dir, "map_metadata_from_infercnv.txt")
+
+# define output infercnv obj
+scratch_obj_file <- file.path(scratch_dir, "run.final.infercnv_obj")
+output_obj_file <- file.path(opts$output_dir, glue::glue("{library_id}_reference-{ref_name}_cnv-obj.rds"))
+
+# png file to save
+scratch_png <- file.path(scratch_dir, "infercnv.png")
+output_png <- file.path(opts$output_dir, glue::glue("{library_id}_reference-{ref_name}_infercnv.png"))
+
+
+# Prepare input data  ---------------------------------------------
+ref_sce <- readRDS(opts$reference_file)
+
+# Update SCE column names to always prefix with library for downstream bookkeeping
+colnames(sce) <- glue::glue("{library_id}-{colnames(sce)}")
+
+# Remove any reference cells which also appear in this library
+duplicated_cells <- intersect(colnames(ref_sce), colnames(sce))
+sce <- sce[, !(colnames(sce) %in% duplicated_cells)]
+
+# Create input matrix by combining library SCE and reference SCE counts assays
+raw_counts_matrix <- cbind(
+  as.matrix(counts(sce)),
+  as.matrix(counts(ref_sce))
+)
+
+# Check that we removed the duplicates:
+stopifnot(
+  "Duplicate cells present in input to inferCNV" = !all(duplicated(colnames(raw_counts_matrix)))
+)
+
+# Define reference group name for inferCNV
+reference_group_name <- "reference"
+
+# Create and export annotation table for inferCNV
+data.frame(
+  cell_id = colnames(sce),
+  annotation = "unknown" # TODO: Should we make this a consensus annotation once that information is in the SCEs? Is this overkill?
+) |>
+  dplyr::bind_rows(
+    data.frame(
+      cell_id = colnames(ref_sce),
+      annotation = reference_group_name
+    )
+  ) |>
```

Comment:
```suggestion
data.frame(
  cell_id = colnames(sce)
  ) |>
  dplyr::mutate(annotations = dplyr::if_else(barcodes %in% normal_cells, "reference", "unknown")
```
I think this is a little clearer to follow. Also I don't think we need to include the annotations since we want to consider all non-reference cells together for CNV inference.